src/CBT.NuGet/Internal/AggregatePackage.cs

```diff
@@ -0,0 +1,141 @@
+            OutPropertyValue = Path.Combine(destinationRoot, string.Format("{0}.{1}", outPropertyId, GetOutputPropertyHash()));
+        }
+
+        public string OutPropertyId { get; private set; }
+
+        public string OutPropertyValue { get; private set; }
+
+        public List<KeyValuePair<AggregateOption, string>> PackagesToAggregate { get; private set; }
+        private string GetOutputPropertyHash()
+        {
+            StringBuilder valueToHash = new StringBuilder();
+            foreach (var pkg in PackagesToAggregate)
+            {
+                valueToHash.Append(pkg.Value.ToLower());
+            }
+            return string.Format("{0:X}", Convert.ToBase64String(Encoding.UTF8.GetBytes(valueToHash.ToString())).GetHashCode());
+        }
+
+        public static AggregatePackage ParseIntoPackage(string input, string pkgDestRoot)
+        {
+            try
+            {
+                // Review split function for something more robust and proper.
+                List<KeyValuePair<AggregateOption, string>> packagesToAggregate = new List<KeyValuePair<AggregateOption, string>>();
+                int aggIndex = 0;
+                var aggPkg = input.Split('=');
+                var pkgID = aggPkg[0];
+                string aggPkgs = aggPkg[1];
+                char[] splitters = new char[] { '*', '|' };
+                aggIndex = aggPkgs.IndexOfAny(splitters);
+                // add base package
+                if (aggIndex < 0)
+                {
+                    aggIndex = aggPkgs.Length;
+                }
+                packagesToAggregate.Add(new KeyValuePair<AggregateOption, string>(AggregateOption.Add, aggPkgs.Substring(0, aggIndex)));
+                aggPkgs = aggPkgs.Substring(aggIndex);
+                aggIndex = aggPkgs.IndexOfAny(splitters);
+                while (aggPkgs.Length > 0)
+                {
+                    char indexChar = aggPkgs[0];
+                    aggPkgs = aggPkgs.Substring(1);
+                    aggIndex = aggPkgs.IndexOfAny(splitters);
+                    if (aggIndex < 0)
+                    {
+                        aggIndex = aggPkgs.Length;
+                    }
+                    if (indexChar.Equals('*'))
+                    {
+                        packagesToAggregate.Add(new KeyValuePair<AggregateOption, string>(AggregateOption.Add, aggPkgs.Substring(0, aggIndex)));
+                    }
+                    if (indexChar.Equals('|'))
+                    {
+                        packagesToAggregate.Add(new KeyValuePair<AggregateOption, string>(AggregateOption.Remove, aggPkgs.Substring(0, aggIndex)));
+                    }
+                    aggPkgs = aggPkgs.Substring(aggIndex);
+                }
+
+                return new AggregatePackage(pkgID, packagesToAggregate, pkgDestRoot);
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        public bool CreateAggregatePackage()
```

Comment:
Yea I don't disagree with that.  Is there a clean pattern you can think of to do logging with out having to make this AggregatePackage object aware of the msbuild logger by passing it in.  Perhaps in this case that is a little over engineering.